src/cmap/connection.ts

```diff
@@ -1208,11 +1150,17 @@ const kDefaultMaxBsonMessageSize = 1024 * 1024 * 16 * 4;
  * Note that `for-await` loops call `return` automatically when the loop is exited.
  */
 export async function* readWireProtocolMessages(
-  connection: ModernConnection
+  connection: ModernConnection,
+  { signal }: { signal?: AbortSignal } = {}
 ): AsyncGenerator<Buffer> {
   const bufferPool = new BufferPool();
   const maxBsonMessageSize = connection.hello?.maxBsonMessageSize ?? kDefaultMaxBsonMessageSize;
-  for await (const [chunk] of on(connection.socket, 'data')) {
+  for await (const [chunk] of on(connection.socket, 'data', { signal })) {
+    if (connection.delayedTimeoutId) {
```

Comment:
can this live in `connection.command` instead?

Reply:
I guess I'd say that if we want this to be "closer to the data event", then `delayedTimeoutId` should not be associated with the connection class?

but we can look at this later.